### PR TITLE
fixes weak ion shots... being stronger than normal ions

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Prevents weak ions from calling the parent on hit effect of ions, causing weak ions to do normal ions effect as well as itself.

## Why It's Good For The Game

Weak ion shots should not... call the normal ion effect as well. This is a bug. This affects HOS gun, and ion scatter rounds / mech ion shotgun.

This does nerf them greatly, but is a bug fix. If the effects of these weapons are too weak, we can change them to normal ion rounds.

For comparison, ion scatter rounds point blank did 180 ish (rng might make it lower, going off reading code / memory rather than a test) energy damage to mechs. (not counting battery drain)
It now does 60.

HOS gun would do 45 to a mecha, it now does 15.

## Testing

Spawn in.
Fire ion rifle, works as intended.
Fire hos gun, works as intended.
Fire ion scatter rounds, works as intended.

## Changelog
:cl:
fix: Fixes weak ion projectiles calling the emp effect from normal ion projectiles as well.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -7,18 +7,18 @@
...
+	var/weak = FALSE
 
 /obj/item/projectile/ion/on_hit(atom/target, blocked = 0)
 	..()
-	empulse(target, 1, 1, 1, cause = "[type] fired by [key_name(firer)]")
+	if(!weak)
+		empulse(target, 1, 1, 1, cause = "[type] fired by [key_name(firer)]")
+	else
+		empulse(target, 0, 0, 1, cause = "[type] fired by [key_name(firer)]")
 	return 1
```
Comment (hal9000PR) [inline code/modules/projectiles/projectile/special.dm:18]: would you instead make ion projectiles have 2 variables, one for light emp, one for heavy EMP, and override it on the weak one?
also change the fourth param to a TRUE

Comment (Qwertytoforty) [reply code/modules/projectiles/projectile/special.dm:18]: sure, give me a bit

Comment (Qwertytoforty) [reply code/modules/projectiles/projectile/special.dm]: done!

**Comment by Dragonkiller93** [comment]
Could you change the cl so it's more explicit in the player sided effects? I.E "Fixes EMP, significant nerf to HOS ion and mech ion."

It's accurate as is but players can't tell that stuff has been hard nerfed, which is the more important part.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**